### DISCUSSION TITLE
Help needed 🙏  autocompletion for local vars and LSP stopped working last week

### DISCUSSION THREAD

**Discussion by asfaltboy** [LazyVim/LazyVim#3897] [Q&A] [answered]
My config includes the language extras for a number of languages, and auto-complete for them worked fine until last week.

Since last week, some of the auto-complete sources stopped working. Lazy still shows cmp-buffer, cmp-nvim-lsp plugins as loaded, but their suggestions are not displayed.

Snippets completions do work, and if I add copilot then both snippets and copilot work, but nothing else is shown.

Running `:CmpStatus` shows:

```
# ready source names
- snippets
# unused source names
- vim-dadbod-completion
# unknown source names
- nvim_lsp
- path
- buffer
- lazydev
```

I don't know for sure if this indicates an issue with nvim_lsp or the buffer source.. I would love some help 🙏

**Comment by folke** [comment]
What does `:LspInfo` show?

**Reply by asfaltboy** [reply]
It seems to be fine:

```
 Language client log: /Users/pavel.savchenko/.local/state/nvim/lsp.log
 Detected filetype:   go
 
 2 client(s) attached to this buffer: 
 
 Client: copilot (id: 1, bufnr: [5, 103])
 	filetypes:       
 	autostart:       false
 	root directory:  /Users/pavel.savchenko/Code/project/modules
 	cmd:             node /Users/pavel.savchenko/.local/share/nvim/lazy/copilot.lua/copilot/index.js
 
 Client: gopls (id: 2, bufnr: [5, 103])
 	filetypes:       go, gomod, gowork, gotmpl
 	autostart:       true
 	root directory:  /Users/pavel.savchenko/Code/project/modules
 	cmd:             /Users/pavel.savchenko/.local/share/nvim/mason/bin/gopls
 
 Configured servers list: zls, gopls, cssls, marksman, sqlls, terraformls, ruff, dockerls, pyright, ruff_lsp, vimls, denols, htmx, rust_analyzer, prismals, pylsp, lua_ls, html, docker_compose_language_service, vtsls, eslint, jsonls, tflint
```

I should mention that using other LSP features like go to definition, type info on Hover, and references work.

**Reply by asfaltboy** [reply]
I just noticed that the cmp docs say that ["unknown" status for some sources, like cmp-nvim-lsp is fine](https://github.com/hrsh7th/nvim-cmp/blob/a110e12d0b58eefcf5b771f533fc2cf3050680ac/doc/cmp.txt#L374-L378), but I don't know how to understand if the lsp source works (not without checking that plugin out and adding logging [in here](https://github.com/hrsh7th/cmp-nvim-lsp/blob/39e2eda76828d88b773cc27a3f61d2ad782c922d/lua/cmp_nvim_lsp/init.lua#L101))

**Reply by folke** [reply]
sorry, no idea then

**Comment by dpetka2001** [comment]
You should try with default LazyVim installation and see if that occurs. If not you're probably doing something in your personal configuration (maybe overwrite some `nvim-cmp` configuration?). 

For me, in both Python and Lua, it shows 
```
# ready source names
- buffer
- path
- snippets
- lazydev
- nvim_lsp:lua_ls  (here it shows pyright instead of lua_ls in case of Python)
# unavailable source names
- git
```

**Reply by asfaltboy** [reply]
I tried on another machine, with the same config (I mange the nvim config dir with my dotfiles) and the completion using buffer and lsp works fine.

Then I uninstalled neovim and lazyvim, and reinstalled both from scratch, with same last working config and no luck. I'll try to create a minimal build with no additional config.

**Reply by asfaltboy** [reply]
So that worked! Without any plugins the cmp-buffer plugin works fine, and when installing the language extra, so does the cmp-nvim-lsp it seems. So now I need to find the culprit, I will add the plugins/config back one by one and see if and when it breaks

**Reply by asfaltboy** [reply, marked as answer]
The issue was this: https://github.com/kristijanhusak/vim-dadbod-completion/issues/53

I just noticed that sql extra includes the workaround from @fredrikaverpil , and a few other improvements to my config, so I switched to using it.
